### PR TITLE
[T3A3][T09-A3]Francis Cheng

### PR DESCRIPTION
**Pull request by francischeng070**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -8,20 +12,39 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String EXAMPLE = "123, #12-645, some street, 213551";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should have Block, Unit, Street and Postal Code";
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: how are you enforcing this format?

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -8,20 +12,39 @@
...
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
     public final String value;
     private boolean isPrivate;
+    private Block block;
+    private Unit unit;
+    private Street street;
+    private PostalCode postalCode;
+    private final addressLimit = 4;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: if you intend to make this like a "constant", the variable name must be in all caps according to the coding standards we follow

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -8,20 +12,39 @@
...
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
+        
         String trimmedAddress = address.trim();
+        String[] splitAddress = trimmedAddress.split(", ");
+        /**
+         * Prevent array out of bound when assigning values
+         */
+        if(splitAddress.length==addressLimit) {
+            block = new Block(splitAddress[0]);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: why 0? why not 1 or 2 or 100? :-P
I hope you understand what I mean!

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -8,20 +12,39 @@
...
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
+        
         String trimmedAddress = address.trim();
+        String[] splitAddress = trimmedAddress.split(", ");
+        /**
+         * Prevent array out of bound when assigning values
+         */
+        if(splitAddress.length==addressLimit) {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: need space around `==`

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -8,20 +12,39 @@
...
+            block = new Block(splitAddress[0]);
+            unit = new Unit(splitAddress[1]);
+            street = new Street(splitAddress[2]);
+            postalCode = new Postalcode(splitAddress[3]);
+        }else{
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: need space around `else`

File: src/seedu/addressbook/data/person/Address.java [hunk 6]
```diff
@@ -45,7 +68,7 @@ public String toString() {
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                        && this.value.equals(((Address) other).value)); // state check
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: hmmmm, why do you need to still retain `value`?
think what are the potential problems of retaining the value field as it is?

File: src/seedu/addressbook/data/person/Block.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package seedu.addressbook.data.person;
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: missing header comments!

when you create a new class, you need to let users and developers know what is its purpose.
hence, need a crisp header comment

**Review by okkhoy — COMMENTED** [review]
please note that you need  to update the documentation to reflect these changes you made to the code (in program help and user guide).
also, you need to update the test files to test the new behavior of splitting address to components!

**Comment by okkhoy** [comment]
@francischeng070 some comments added, please check.
please close the PR once you have noted the suggestions

File: src/seedu/addressbook/data/person/Address.java (hunk 3 above)
Comment (francischeng070) [reply src/seedu/addressbook/data/person/Address.java:40]: ok i understand

**Comment by francischeng070** [comment]
ack

**Pull request closed without merging**